### PR TITLE
#125 refactor sparql js

### PR DESCRIPTION
**Pull request by roryschadler**
fix (#125)

Added a combined retrieve and parse function for SPARQL queries, and made both retrieval and parsing fail noisily instead of quietly.

### CODE REVIEW

**Comment by roryschadler** [comment]
Meant to submit this on Wednesday morning, oops! @aswallace was this what you had in mind?

**Review by aswallace — COMMENTED** [review]
Yep! Would also be good to get opinion of @stouffers who brought this issue up originally

**Comment by kevinzunigacuellar** [comment]
Don't forget to throw errors for HTTP responses 4xx and 5xx [MDN fetch api](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)

**Comment by roryschadler** [comment]
> Don't forget to throw errors for HTTP responses 4xx and 5xx [MDN fetch api](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)

Good catch, I forgot those didn't throw errors by default.

File: app/src/modules/sparql.js [hunk 1]
```diff
@@ -34,8 +39,14 @@ function parseSparql (response) {
         rowData[field] = value
       })
     }
+  } catch (err) {
+    throw new Error('Error parsing response', { cause: err })
```
Comment (tholulomo) [inline app/src/modules/sparql.js:43]: I'll suggest passing the `err` itself as the argument and then removing the object in both cases.

**Review by tholulomo — APPROVED** [review]

**Pull request merged**